Bounties-Network/BountiesAPI
# New email templates

✉️ @villanuevawill @c-o-l-o-r  this one's a doozie! ✉️ 

So, in the past we've used one base template for all our email notifications. It was simple and got the job done - a button to go back to the site and a sentence explaining what happened. 

Moving forwards, we have many known events and some unknown events - they are appearing as we refine the experience of both the issuer and the fulfiller. 

Each of these user types receive notifications about each others' actions on bounties they're involved with. 

The from / to is handled in a very simple way in the current system, making it difficult to reliably pass one value through the event leading up to the email being sent. 

With all these new notifications, we need a way to pass events through without fiddling with them first. To relate to this message, I create an `Email` class that determines what goes in the "from" fields of the email templates, and what goes in the "to" fields. 

The end result is a simplified notification handler, but more logic in the Email class encapsulated away from the sending of the original notification. 

I'd love to hear your thoughts on the structure and how useable it will be moving forwards into more events. 

This pattern should work well in a potentially functional future infrastructure, splitting away from other code and living on its own as an email template renderer. 


## Several todo's left that will wrap up quickly

- [x] Calculate usd the best way
- [ ] Stop writing html files to local disk
- [x] Update link to correct domain for settings
- [ ] Test with profile images
- [x] Re-enable other notifications
- [x] Fetch pending submissions and use that instead of remaining funds divided by fulfillment cost?
- [x] Truncate the address in the middle
- [x] Rating
- [x] Notify for comments to everyone who has commented on that same bounty
- [x] Event for creating and updating a draft bounty (notification happens front end)

## How to test

Simply fire up `docker-compose up` locally and see `bounties_api/` fill up with the notifications that would be sent out. If you have a nice terminal you can type `git status` and then `cmd+click` the html files that show up to open them in your browser to see how the rendered notification looks. Be sure to delete those files afterwards :)
